pkg/segment/reader/record/rrcreader.go

```diff
@@ -224,6 +224,7 @@ func GetJsonFromAllRrc(allrrc []*utils.RecordResultContainer, esResponse bool, q
 		} else {
 			recs = make(map[string]map[string]interface{})
 			finalCols = nodeRes.FinalColumns
+			nodeRes.ColumnsOrder = colsIndexMap
```

Comment:
nit: now you're doing `nodeRes.ColumnsOrder = colsIndexMap` in both the `if` (line 216) and the `else`, so you should move it out to above/below the `if/else`

Reply:
addressed in the latest commit.